### PR TITLE
Address JavaDoc errors/warnings in some packages

### PR DESCRIPTION
**Pull request by thc202**
Address JavaDoc errors/warnings (Java 7 and 8) in packages "utils",
"authentication", "spider" and "users", for example:
 - warning: no description for `@param`;
 - warning: no description for `@throws`;
 - warning: no `@return`;
 - warning: no `@param` for ...;
 - error: self-closing element not allowed;
 - error: unexpected text (in see tags);
 - error: reference not found;
 - error: unknown tag: ...;
 - error: `@param` ... not found.

### CODE REVIEW

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java [hunk 1]
```diff
@@ -149,22 +151,23 @@ public abstract AuthenticationMethod loadMethodFromSession(Session session, int
...
 	 */
 	public abstract void persistMethodToSession(Session session, int contextId,
-			AuthenticationMethod authMethod) throws UnsupportedAuthenticationMethodException, DatabaseException;
+			AuthenticationMethod authMethod) throws DatabaseException;
 
 	/**
 	 * Export the specified method to the configuration
-	 * @param config
-	 * @param authMethod
+	 * @param config the configurations where to export/save the authentication method
```
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: The config is the save location?

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java [hunk 2]
```diff
@@ -149,22 +151,23 @@ public abstract AuthenticationMethod loadMethodFromSession(Session session, int
...
 	 */
 	public abstract void exportData(Configuration config, AuthenticationMethod authMethod);
 
 	/**
-	 * Import the method from the confuguration
-	 * @param config
-	 * @param authMethod
-	 * @throws ConfigurationException
+	 * Import the method from the configuration
+	 * @param config the configurations from where to import/load the authentication method
```
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: This one is confusing too...

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java [hunk 3]
```diff
@@ -149,22 +151,23 @@ public abstract AuthenticationMethod loadMethodFromSession(Session session, int
...
+	 * @param authMethod where to set the imported authentication method data
```
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: The method provides location?

File: src/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodType.java [hunk 4]
```diff
@@ -273,6 +274,7 @@ public WebSession authenticate(SessionManagementMethod sessionManagementMethod,
 		 * Sets the login request as being an existing SiteNode.
 		 * 
 		 * @param loginSiteNode the new login request
+		 * @throws Exception if an occurred while obtaining the message from the node
```
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodType.java]: "If one occurred"?

File: src/org/zaproxy/zap/spider/SpiderParam.java [hunk 5]
```diff
@@ -696,8 +696,9 @@ public void setHandleParameters(HandleParametersOption handleParametersVisited)
 	 * {@code "USE_ALL"}, {@code "IGNORE_VALUE"}, {@code "IGNORE_COMPLETELY"}.
 	 * 
 	 * @param handleParametersVisited the new handle parameters visited value
+	 * @throws IllegalArgumentException if the given parameter is not a value of {@code HandleParametersOption}.
```
Comment (kingthorin) [inline src/org/zaproxy/zap/spider/SpiderParam.java]: The throws declaration was removed below, and the code doesn't do a throw?

File: src/org/zaproxy/zap/spider/SpiderTask.java [hunk 6]
```diff
@@ -311,9 +311,9 @@ private ExtensionHistory getExtensionHistory() {
 	 * @return the response http message
 	 * @throws HttpException the http exception
```
Comment (kingthorin) [inline src/org/zaproxy/zap/spider/SpiderTask.java]: This declaration was removed below?

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java [hunk 7]
```diff
@@ -26,7 +26,7 @@
  * or the similarity ratio between two strings.
  * This implementation is using Hirschber's algorithm B and algorithm C.
  *
- * @see <a href="http://code.google.com/p/algorithm800/source/browse/trunk/src/Hirschberg.java">Hirschberg&apos;s algorithm
+ * @see <a href="http://code.google.com/p/algorithm800/source/browse/trunk/src/Hirschberg.java">Hirschberg's algorithm
```
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Should we update this? https://code.google.com/archive/p/algorithm800/source/default/source

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java [hunk 8]
```diff
@@ -41,15 +41,6 @@
     // Maximum value for comparison ratio
     public static final double MAX_RATIO = 1.0;
     
-    /**
-     * Algorithm B as described by Hirschberg
```
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Is this and the similar C comment below worth leaving, even as a single line comment for future maintenance,  or do you think the function name is enough?

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java [hunk 9]
```diff
@@ -123,23 +105,14 @@ private static void algC(StringBuilder sb, int m, int n, String a, String b) {
...
+     * @return the reversed string
      */
     private static String reverseString(String in) {
         StringBuilder out = new StringBuilder(in).reverse();
         return out.toString();
     }
 
-    /**
-     * This method finds the index of the maximum sum of L1 and L2, as described
-     * by Hirschberg
```
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Should these two be left in for clarity?

File: src/org/zaproxy/zap/utils/ThreadUtils.java [hunk 10]
```diff
@@ -10,9 +10,9 @@
 	 * like {@link EventQueue#invokeAndWait(Runnable)}, but can be called from the main thread as
 	 * well.
 	 * 
-	 * @param runnable
-	 * @throws InterruptedException
-	 * @throws InvocationTargetException
+	 * @param runnable the {@code Runnable} to be ran in the EDT
```
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/ThreadUtils.java]: Think this should be "run"

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java [hunk 11]
```diff
@@ -149,22 +151,23 @@ public abstract AuthenticationMethod loadMethodFromSession(Session session, int
 	 * @param authMethod the auth method to persist
 	 * @throws UnsupportedAuthenticationMethodException the unsupported authentication method
```
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: The UnsupportedAuthenticationMethodException is an unchecked exception, it's not required to be declared.

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 1 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: Yes, the data of the authentication method is "saved" to the configuration.
Would the following be more clear?
`the {@code Configuration} where to export/save the authentication method`

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 2 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: How about:
`the {@code Configuration} from where to import/load the authentication method`
is it more clear?

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 3 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: No, the `authMethod` is where the imported data (read from the Configuration) is set.

File: src/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodType.java (hunk 4 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/FormBasedAuthenticationMethodType.java]: It's missing "error", will add that.

Yes, `getHttpMessage()` throws some exceptions.

File: src/org/zaproxy/zap/spider/SpiderParam.java (hunk 5 above)
Comment (thc202) [inline src/org/zaproxy/zap/spider/SpiderParam.java]: That exception is not required to be declared. The call `valueOf(String)` might throw that exception (when there's no enum value that corresponds to the passed string).

File: src/org/zaproxy/zap/spider/SpiderTask.java (hunk 6 above)
Comment (thc202) [inline src/org/zaproxy/zap/spider/SpiderTask.java]: That exception is already included by the declaration of IOException, same for other exceptions re-thrown.

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 7 above)
Comment (thc202) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Will change that (although that will also be removed soon?).

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 8 above)
Comment (thc202) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: The method name seems to be enough (the comment does not add anything IMO).

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 9 above)
Comment (thc202) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Is it more clear with that? What is L1, L2 and n?

I was going to document these methods properly but didn't find the paper with the description of the algorithm.

File: src/org/zaproxy/zap/utils/ThreadUtils.java (hunk 10 above)
Comment (thc202) [inline src/org/zaproxy/zap/utils/ThreadUtils.java]: Right, will fix that.

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: Are we talking about disk location, or the details within a config file,  or___?

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 2 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: Are we talking about disk location, or the details within a config file,  or___?

File: src/org/zaproxy/zap/spider/SpiderParam.java (hunk 5 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/spider/SpiderParam.java]: Ok

File: src/org/zaproxy/zap/spider/SpiderTask.java (hunk 6 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/spider/SpiderTask.java]: Ok

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 7 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Well no big deal then, links will die over time people will just have to accept that.

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 9 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Fair enough

File: src/org/zaproxy/zap/utils/HirshbergMatcher.java (hunk 8 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/utils/HirshbergMatcher.java]: Ok

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 1 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: All of that? `Configuration` is an interface so it depends on the implementation used, the data can end up in the memory, the file system or even a remote location.

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 2 above)
Comment (thc202) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: Same as previous comment (but `Configuration` as source of the data instead).

File: src/org/zaproxy/zap/authentication/AuthenticationMethodType.java (hunk 1 above)
Comment (kingthorin) [inline src/org/zaproxy/zap/authentication/AuthenticationMethodType.java]: Oh ok, that makes more sense.

**Comment by thc202** [comment]
Changes done.

**Comment by kingthorin** [comment]
Looks good

**Pull request merged**